### ISSUE TITLE
Fold isn't working, could use some docs, or even just a few code comments on how it's supposed to work

### ISSUE THREAD

**Issue by daveman1010221** [blankname/vim-fish#5]
visual mode range select + zf fails with 'Cannot create fold with current foldmethod', no matter what I set foldmethod to in init.vim. I wish I had more to go on, sorry.

```
NVIM v0.4.4
Build type: Release
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -g -O2 -fdebug-prefix-map=/build/neovim-u4YhjF/neovim-0.4.4=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=1 -DDISABLE_LOG -Wdate-time -D_FORTIFY_SOURCE=1 -O2 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/build/neovim-u4YhjF/neovim-0.4.4/build/config -I/build/neovim-u4YhjF/neovim-0.4.4/src -I/usr/include -I/usr/include/lua5.1 -I/build/neovim-u4YhjF/neovim-0.4.4/build/src/nvim/auto -I/build/neovim-u4YhjF/neovim-0.4.4/build/include
Compiled by team+vim@tracker.debian.org

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

```
:checkhealth

health#coc#check
========================================================================
  - OK: Environment check passed

  - OK: Javascript bundle build/index.js found
  - OK: Service started

health#LanguageClient#check
========================================================================
  - OK: binary found: /home/xxxx/.config/nvim/plugged/LanguageClient-neovim/bin/languageclient
  - OK: languageclient 0.1.161
  - OK: Floating window is supported and will be used for hover

health#nvim#check
========================================================================
## Configuration
  - OK: no issues found

## Performance
  - OK: Build type: Release

## Remote Plugins
  - OK: Up to date

## terminal
  - INFO: key_backspace (kbs) terminfo entry: key_backspace=\177
  - INFO: key_dc (kdch1) terminfo entry: key_dc=\E[3~
  - INFO: $VTE_VERSION='6200'
  - INFO: $COLORTERM='truecolor'

health#provider#check
========================================================================
## Clipboard (optional)
  - OK: Clipboard tool found: xclip

## Python 2 provider (optional)
  - INFO: Using: g:python_host_prog = "/usr/bin/python2"
  - INFO: Executable: /usr/bin/python2
  - INFO: Python version: 2.7.18
  - INFO: pynvim version: 0.4.1 (outdated; from /usr/local/lib/python2.7/dist-packages/neovim)
  - WARNING: Latest pynvim is NOT installed: 0.4.2

## Python 3 provider (optional)
  - INFO: Using: g:python3_host_prog = "/usr/bin/python3"
  - INFO: Executable: /usr/bin/python3
  - INFO: Python version: 3.8.6
  - INFO: pynvim version: 0.4.1 (outdated; from /usr/lib/python3/dist-packages/neovim)
  - WARNING: Latest pynvim is NOT installed: 0.4.2

## Ruby provider (optional)
  - INFO: Ruby: ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-linux-gnu]
  - INFO: Host: /usr/local/bin/neovim-ruby-host
  - OK: Latest "neovim" gem is installed: 0.8.1

## Node.js provider (optional)
  - INFO: Node.js: v13.14.0
  - INFO: Neovim node.js host: /home/xxxx/.node/lib/node_modules/neovim/bin/cli.js
  - OK: Latest "neovim" npm/yarn package is installed: 4.9.0
```

**Comment by blankname** [comment]
Can you check the value of `foldmethod` at runtime when editing a .fish file using:
```:verbose set foldmethod?```

```:help zf```
indicates that `zf` only works when `foldmethod` is set to `manual` or `marker`.

It's possible that another plugin is overriding the `foldmethod` that you're setting in `init.vim`.

**Comment by daveman1010221** [comment]
For some reason, my NeoVim remembers the state of variables if I set them, for example, `:set foldmethod=manual`. This seems to have resolved the issue and folding now works. It was set to foldmethod=diff, which obviously won't work, but was overriding what was written in init.vim. This seems to be set in a ~/.local/share/nvim/view folder, which contains files for literally every place I have ever used my editor, and the files seem to contain localized variable state for every single file. I have no idea how I turned this on, but knowing that this is here is _really_ important. :joy: 

Thanks for the assist.
